src/pattern-library/components/Library.tsx

```diff
@@ -265,6 +265,86 @@ function Demo({
   );
 }
 
+export type LibraryExampleDemoProps = Omit<LibraryDemoProps, 'children'> & {
+  /**
+   * Example file to read and use as content (to be rendered with syntax
+   * highlighting)
+   */
+  exampleFile: string;
+};
+
+function ExampleDemo({
+  exampleFile,
+  classes,
+  title,
+  withSource,
+  style,
+}: LibraryExampleDemoProps) {
+  const [visibleTab, setVisibleTab] = useState('demo');
+  const [children, setChildren] = useState<ComponentChildren>();
+
+  useEffect(() => {
+    import(`../examples/${exampleFile}.tsx`).then(({ App }) => {
+      setChildren(<App />);
+    });
```

Comment:
When importing an example file as a module, this assumes it exports an `App` symbol. We could think on better approaches, perhaps by relaying on the default export instead, as it is less restrictive.

Reply:
Yep, it's a common practice in the framework world (Remix routes and Next.js pages, for example)